test/hardhat/FullMath.spec.ts

```diff
@@ -0,0 +1,214 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { FullMathTest } from "../../typechain-types";
```

Comment:
I don't have such a typechain-types folder on my machine.

Reply:
This is generated when `npx hardhat test` is called.